middlewares/middlewares_test.go

```diff
@@ -0,0 +1,51 @@
+package middlewares
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestGetRemoteAddr(t *testing.T) {
+
+	cases := []struct {
+		name      string
+		header    string
```

Comment:
Since you're setting header every time, I don't think this test will ever catch the change you made. You can check this by running:

```bash
go test -coverprofile=coverage.out ./middlewares/
go tool cover -html=coverage.out
```

You could probably make `header` a `map[string]string` and make set the `ipAddress` field to the `RemoteAddr` field on your request after you create it.

As @ethanfrogers pointed out, it might be good to add some test cases that include header values like `[::]:1234` (ipv6 localhost)